expect/test.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+import { expect } from "./expect.ts";
+import { AssertionError, assertRejects, assertThrows } from "@std/assert";
+
+Deno.test("expect().resolves.toEqual()", async () => {
+  await expect(Promise.resolve(42)).resolves.toEqual(42);
+  await expect(Promise.resolve(42)).resolves.not.toEqual(0);
+
+  assertThrows(
+    () => expect(42).resolves.toEqual(42),
+    AssertionError,
+    "expected value must be Promiselike",
+  );
+  assertThrows(
```

Comment:
Just checking that we want this to throw rather than reject, like Jest?

Reply:
Yes. I checked with the below jest snippets just in case. They throw instead of returning rejected promise.

```ts
describe("mod", () => {
  it("func", () => {
    try {
      expect(null).resolves.toEqual(null);
    } catch (e) {
      console.log("thrown", e);
    }
    try {
      expect(42).resolves.toEqual(42);
    } catch (e) {
      console.log("thrown", e);
    }
  });
});
```